site/develop/index.md

```diff
@@ -67,6 +67,26 @@ helps to agree on the general approach. Log a
 [JIRA case](https://issues.apache.org/jira/browse/CALCITE) for your
 proposed feature or start a discussion on the dev list.
 
+Before opening up a new JIRA case, have a look in the existing issues.
+The feature or bug that you plan to work on may already be there.  
+
+If you are going to take on the issue right away assign it to your self.
+To assign issues to your self you have to be registered as a contributor 
+to the project. In order to do that, send an email to the developers list
+and provide your JIRA username.   
+
+If you think the issue can make it to the ongoing release set the fix version
+accordingly (e.g., 1.20.0), otherwise leave it as blank.
+
+If you pick up an existing issue and start working on it, set the fix 
```

Comment:
I'd strike this paragraph. If you pick up an existing issue, mark it 'in progress', and when it's finished flag 'pull-request-available'. You never know whether you will finish in time, so there's no point making promises.

Frankly, it's pretty rare that anyone besides the release manager would set fixVersion. Sometimes if pull-request-available is set, a reviewer will set fixVersion to remind them to review before the release.

Reply:
Agreed.